nomad/job_endpoint_hook_implicit_identities.go

```diff
@@ -75,19 +70,25 @@ func (h jobImplicitIdentitiesHook) handleConsulService(s *structs.Service) {
 
 	// Set the expected identity name and service name.
 	name := s.MakeUniqueIdentityName()
-	serviceWID.Name = fmt.Sprintf("%s/%s", consulServiceIdentityNamePrefix, name)
+	serviceWID.Name = fmt.Sprintf("%s_%s", structs.ConsulServiceIdentityNamePrefix, name)
 	serviceWID.ServiceName = s.Name
 
 	s.Identity = serviceWID
 }
 
-func (h jobImplicitIdentitiesHook) handleConsulTasks(t *structs.Task, taskGroup string) {
+func (h jobImplicitIdentitiesHook) handleConsulTasks(t *structs.Task) {
 	if !h.srv.config.UseConsulIdentity() {
 		return
 	}
 
-	name := t.MakeUniqueIdentityName(taskGroup)
-	widName := fmt.Sprintf("%s/%s", consulTaskIdentityNamePrefix, name)
+	var clusterName string
+	if t.Consul != nil && t.Consul.Cluster != "" {
+		clusterName = t.Consul.Cluster
+	} else {
+		clusterName = structs.ConsulDefaultCluster
+	}
+
+	widName := fmt.Sprintf("%s_%s", structs.ConsulTaskIdentityNamePrefix, clusterName)
```

Comment:
I have this in a helper method to avoid repeating the logic here and in the alloc/task runner
https://github.com/hashicorp/nomad/blob/3633ca0f8ca4c0a7d4ecb43840e1240c10f45116/nomad/structs/structs.go#L10000-L10004